docs/guides/publish_fetch_packages.md

```diff
@@ -0,0 +1,177 @@
+
+Packages live in a different space when being used by the developer, and they are published in a registry using different commands. The table below presents a summary.
+
+| Package type | Location while being used           | Command to publish in a registry | Command to retrieve from a registry |
+|--------------|-------------------------------------|----------------------------------|-------------------------------------|
+| Service      | Independent folder                  | `autonomy publish`               | `autonomy fetch`                    |
+| Agent        | Independent folder                  | `autonomy publish`               | `autonomy fetch`                    |
+| Component    | `/vendor` folder inside agent folder | `autonomy push`                  | `autonomy add`                      |
+
+
+
+The figure below shows a typical setup, where you might identify some common traits with the Git methodology. Within the workspace folder, the arrows in the diagram point to the locations where the different operations can be executed in the local machine.
+
+<figure markdown>
+![](../images/package_management.svg){ width="85%" height="85%" style="display: block; margin: 0 auto" }
+<figcaption>Overview of the package management flow with the Open Autonomy framework</figcaption>
+</figure>
+
+!!! warning "Important"
+    Do not confuse the **local registry** with a local **workspace folder**. The local registry is simply a repository to store finalized packages, mimicking the role of a remote registry on the local machine.
+
+    For example, you can develop a component package directly in the local registry, or you can develop it in an agent within a workspace folder, and push the component later in the local registry. Then, you can retrieve that component from the registry for use in another agent.
+
+    **Regardless of the methodology that you choose to develop your packages, it is important to have a clear separation of concerns in mind so that they can be reused in other agent projects.**
+
+## How to tell the framework what registry to use
+The `push`, `add`, `publish` and `fetch` commands use, by default, the registry specified when the framework was initiated (command `autonomy init`). See for example the [set up guide](./set_up#set-up), where we initialized the framework to use the default remote [IPFS](https://ipfs.io) registry.
+
+Additionally, the framework configuration can be overridden per command by using the flags `--local` or `--remote` in any of the commands  `push`, `add`, `publish` or `fetch`.
+
+
+## Push and add components
+This section assumes that you have a newly created component, e.g., an {{fsm_app}} skill, within an agent folder. It is out of the scope of this guide how to create agent components. You can review the [guide to create an {{fsm_app}}](./create_fsm_app.md), or the {{open_aea_doc}}. Your component should be located within an agent folder (i.e., a folder containing the file `aea-config.yaml`) in
```

Comment:
Are we refering here to an independent agent folder or to one in the  local registry?

Reply:
Independent agent folder. See comment below.